### DISCUSSION TITLE
Unresponsive services in ROS2 - not fixed

### DISCUSSION THREAD

**Discussion by esharet** [eclipse-zenoh/roadmap#104] [Q&A] [answered]
HI there.
I am trying to set up a ROS2 service in one ROS domain, and to send a request from another domain ID. I'm trying to connect them using 2 bridges of DDS2ROS.
**The request is reaching to the server, but I couldnt recieve the reply back to the client.**

terminal 1:
ROS_DOMAIN_ID=1 ros2 run demo_nodes_py add_two_ints_client

terminal 2:
ROS_DOMAIN_ID=2 ros2 run demo_nodes_py add_two_ints_server

terminal 3:
zenoh-bridge-dds -d 1

terminal 4:
zenoh-bridge-dds -d 2

Even though it was written in multiple places that this bug was fixed (bug fixed in 102, 108 in dds-bridge), I failed making it work.
I believe that I don't use the bridge correctly in the case of services... maybe something with the configuration file of --allow? 

Appreciate your help.

**Comment by JEnoch** [comment, marked as answer]
Hi @esharet ,

Please see my answer to a very similar question here: https://github.com/eclipse-zenoh/zenoh-plugin-dds/issues/149#issuecomment-1830371121

TL;DR: Try the new Zenoh bridge for ROS 2: https://github.com/eclipse-zenoh/zenoh-plugin-ros2dds
